base/threadingconstructs.jl

```diff
@@ -151,11 +159,12 @@ function threading_run(fun, static)
         else
             # TODO: this should be the current pool (except interactive) if there
             # are ever more than two pools.
-            ccall(:jl_set_task_threadpoolid, Cint, (Any, Int8), t, _sym_to_tpid(:default))
+            @assert ccall(:jl_set_task_threadpoolid, Cint, (Any, Int8), t, _sym_to_tpid(:default)) == 1
         end
         tasks[i] = t
         schedule(t)
     end
+    ct.static = ct_static
```

Comment:
This is a behavior change, which probably shouldn't be in the same PR as a bugfix (we should backport this PR).
```suggestion
```

Reply:
Or rather, I suppose it is not really a behavior change, since it should be NFC, but still therefore not needed

Reply:
Yeah I bunched up a couple of changes. I will separate them out.

Reply:
Actually after checking `enq_work` the behavior I was worried about does not happen.